docs: add instructions for custom SVG icons (#1490)
pydata/pydata-sphinx-theme: docs/user_guide/header-links.rst

If you'd like to display an icon image that is not in the FontAwesome icons library,
you may instead specify a URL or a path to a local image that will be used for the icon.
You may also use ``.svg`` images as if they were FontAwesome with a little additional setup.

Bitmap image icons
~~~~~~~~~~~~~~~~~~

For all bitmap image icons such as ``.png``, ``.jpg``, etc., you must specify ``type`` as local.

> .. note:: Note regarding :literal:`.svg`'s

Reviewer: ```suggestion
.. note::
```

note directives don't allow a title

If you'd like to display an icon image that is not in the FontAwesome icons library,
you may instead specify a URL or a path to a local image that will be used for the icon.
You may also use ``.svg`` images as if they were FontAwesome with a little additional setup.

Bitmap image icons
~~~~~~~~~~~~~~~~~~

For all bitmap image icons such as ``.png``, ``.jpg``, etc., you must specify ``type`` as local.

> .. admonition:: Note regarding :literal:`.svg`'s
>     :class: note

Reviewer 2: ```suggestion
.. note:: Note regarding :literal:`.svg`'s
```

   Use ``.svg`` images for a higher-resolution output that behaves similarly across screen sizes.

.. _svg-image-icons:

SVG image icons
~~~~~~~~~~~~~~~

In order to make use of the full feature set of ``.svg`` images provided by HTML you will need
to set up the ``.svg`` to be used as a FontAwesome type icon. This is a fairly straightforward process:

#. Copy the contents of ``custom-icon.js`` - located within the ``docs`` tree - into an appropriate directory of your documentation
   source (typically ``source/js``) and rename the file however you like. Highlighted below are the lines which must be modified

>    .. code:: javascript

Reviewer 2: ```suggestion
   .. code:: javascript

```

   Use ``.svg`` images for a higher-resolution output that behaves similarly across screen sizes.

.. _svg-image-icons:

SVG image icons
~~~~~~~~~~~~~~~

In order to make use of the full feature set of ``.svg`` images provided by HTML you will need
to set up the ``.svg`` to be used as a FontAwesome type icon. This is a fairly straightforward process:

#. Copy the contents of ``custom-icon.js`` - located within the ``docs`` tree - into an appropriate directory of your documentation
   source (typically ``source/js``) and rename the file however you like. Highlighted below are the lines which must be modified

   .. code:: javascript

>    

Reviewer 2: ```suggestion

```

>    .. code:: javascript
>    
>     prefix: "fa-custom",
>     iconName: "pypi",
>     icon: [
>       17.313, // viewBox width
>       19.807, // viewBox height
>       [], // ligature
>       "e001", // unicode codepoint - private use area
>       "m10.383 0.2-3.239 ...", // string definined SVG path
>     ],

Reviewer 2: ```suggestion
   .. code:: javascript
   
     prefix: "fa-custom",
     iconName: "pypi",
     icon: [
       17.313, // viewBox width
       19.807, // viewBox height
       [], // ligature
       "e001", // unicode codepoint - private use area
       "m10.383 0.2-3.239 ...", // string definined SVG path
     ],
```

   Use ``.svg`` images for a higher-resolution output that behaves similarly across screen sizes.

.. _svg-image-icons:

SVG image icons
~~~~~~~~~~~~~~~

> In order to make use of the full feature set of ``.svg`` images provided by HTML you will need
> to set up the ``.svg`` to be used as a FontAwesome type icon. This is a fairly straightforward process:

Reviewer: ```suggestion
If you want to use advanced ``.svg`` features such as motion or dynamic styling,
you can set up the ``.svg`` as a custom FontAwesome type icon. This is a fairly straightforward process:
```

#. Copy the contents of ``custom-icon.js`` - located within the ``docs`` tree - into an appropriate directory of your documentation
   source (typically ``source/js``) and rename the file however you like. Highlighted below are the lines which must be modified

>    .. sourcecode:: javascript
>     :emphasize-lines: 3,5,6,9
> 
>     ...

Reviewer 2: ```suggestion
   .. code:: javascript
   
```

    prefix: "fa-custom",
    iconName: "pypi",
    icon: [
      17.313, // viewBox width
      19.807, // viewBox height
      [], // ligature
      "e001", // unicode codepoint - private use area
      "m10.383 0.2-3.239 ...", // string definined SVG path
    ],

>     ...

Reviewer 2: ```suggestion
```

#. Update the following file contents:

>    #.  ``iconName``  to be one of our choosing

Reviewer: ```suggestion
   #.  ``iconName``  to be one of your choosing
```

   Use ``.svg`` images for a higher-resolution output that behaves similarly across screen sizes.

.. _svg-image-icons:

SVG image icons
~~~~~~~~~~~~~~~

In order to make use of the full feature set of ``.svg`` images provided by HTML you will need
to set up the ``.svg`` to be used as a FontAwesome type icon. This is a fairly straightforward process:

#. Copy the contents of ``docs/_static/custom-icon.js`` into an appropriate directory of your documentation
   source (typically ``source/js``) and rename the file however you like.

   .. literalinclude::  ../_static/custom-icon.js
    :language: javascript
    :emphasize-lines: 7, 13

>     :linenos:

Reviewer 2: Would it be possible to copy/paste only the meaningful content in a standard code-block ? We cannot guarantee this file will remain as it is forever and I would prefer the documentation not to depend on our internal cuisine.
